src/app/ngrx/actions/index.ts

```diff
@@ -1,19 +1,18 @@
-import { RouterStateSnapshot } from '@angular/router';
-import { RouterNavigationPayload } from '@ngrx/router-store';
-import { CmsPodcastsPayload, CmsEpisodeGuidsPayload } from './cms.action.creator';
-import { CastleFilterPayload, CastlePodcastMetricsPayload, CastleEpisodeMetricsPayload } from './castle.action.creator';
-import { GoogleAnalyticsEventPayload } from './google-analytics.action.creator';
+import { RouterNavigationAction } from '@ngrx/router-store';
+import { CmsPodcastsAction, CmsAllPodcastEpisodeGuidsAction } from './cms.action.creator';
+import { CastleFilterAction, CastlePodcastMetricsAction, CastleEpisodeMetricsAction } from './castle.action.creator';
+import { GoogleAnalyticsEventAction } from './google-analytics.action.creator';
 
-export type All
-  = CmsPodcastsPayload
-  | CmsEpisodeGuidsPayload
-  | CastleFilterPayload
-  | CastlePodcastMetricsPayload
-  | CastleEpisodeMetricsPayload
-  | RouterNavigationPayload<RouterStateSnapshot>
-  | GoogleAnalyticsEventPayload;
+export type AllActions
```

Comment:
I _should_ be able to use this discriminating union in the reducers rather than having to use the individual Action classes. The problem is that TypeScript only seeing the properties on the CmsPodcastsAction when you give the reducer's action parameter the type `AllActions`. They are still recommending this though. I was using a later version of TypeScript at one point, but I downgraded to 2.4.0 because that's the earliest that Ngrx supports and Angular doesn't think it supports > 2.4.0.